### ISSUE TITLE
Can't use custom lint rules with Android Gradle Plugin 4.1.0

### ISSUE THREAD

**Issue by bartek-wesolowski** [getsentry/sentry-android-gradle-plugin#31]
_Android Gradle Plugin:_
**4.1.0**

_Gradle:_
**6.5 & 6.7**

_Optimizer:_
- [ ]  ProGuard
- [x]  R8

_Have you created the sentry.properties file?_
- [x]  Yes -> in the main project folder; the integration was working before with AGP 4.0.x
- [ ]  No

_SDKs:_
- [x]  sentry-android -> 1.7.36
- [ ]  sentry-react-native

---
I have the following issue:

I use custom lint rules in my project. I do it by adding the following line in my app/build.gradle:
`lintChecks project(':checks')`
After upgrading to Android Gradle Plugin 4.1.0, when I run "sync with Gradle" then it fails with the following error:
```
A problem occurred evaluating project ':checks'.
> Failed to apply plugin [id 'kotlin']
   > Gradle#projectsEvaluated(Action) on build 'android-studio-4' cannot be executed in the current context.
```
The problem disappears when I either remove custom lint rules or remove the Sentry plugin or set `minifyEnabled false`.

**Steps to reproduce:**
I forked the official Custom Lint Rules repository, added the Sentry plugin to it and set `minifyEnabled true`. Now, trying to sync the project with Gradle fails.
https://github.com/bartek-wesolowski/android-custom-lint-rules/tree/sentry/android-studio-4
Note that the sample doesn't include `sentry.properties`, because I don't want to expose my private data, but I do have this file in my project in the correct spot, because the Sentry integration was already working.

**Comment by marandaneto** [comment]
thanks for raising this.

could you add `org.gradle.logging.level=warn` to gradle.propeties and copy/paste?
we don't do anything with custom lint rules, and the latest version of the plugin also works on AGP 4.1.x

I'll get some time this week to run your sample, that is handy.

**Comment by bartek-wesolowski** [comment]
Here's the interesting fragment from idea.log when I try to sync the sample project with Gradle:
https://gist.github.com/bartek-wesolowski/36e91d0fbb226fd8cef8a2deaa2c8def

**Comment by fredydoss** [comment]
> Here's the interesting fragment from idea.log when I try to sync the sample project with Gradle:
> https://gist.github.com/bartek-wesolowski/36e91d0fbb226fd8cef8a2deaa2c8def

I face a similar issue on my custom lint project. It's not related to sentry but the issue is the same.

**Comment by bartek-wesolowski** [comment]
@fredydoss I'm seeing the same issue with New Relic. I created an issue there as well. Perhaps those plugins use some Gradle instructions incorrectly.

**Comment by neworld** [comment]
We are facing the same problem and I found an issue on google issue tracker: https://issuetracker.google.com/issues/170656529. Looks like it is AGP or kotlin problem

**Comment by int02h** [comment]
We faced the same issue too. It seems that all modules with custom lint checks are affected.

**Comment by int02h** [comment]
You can workaround this issue by including module with custom lint rules only if `System.getProperty("idea.active") != "true"`

**Comment by marandaneto** [comment]
@int02h thanks for the quick workaround.

**Comment by marandaneto** [comment]
yeah looks like an AGP/IDE issue rather than the Sentry gradle plugin

**Comment by marandaneto** [comment]
it was an AGP issue and its been fixed on AGP >= 4.2, See https://issuetracker.google.com/issues/170656529
